presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -345,6 +345,13 @@ public PlanWithProperties visitJoin(JoinNode node, Void context)
                     left.getProperties());
         }
 
+        private boolean isJoinOnPartitionKeys(PlanWithProperties left, List<Symbol> leftSymbols, PlanWithProperties right, List<Symbol> rightSymbols)
+        {
+            return left.getProperties().isPartitioned() && right.getProperties().isPartitioned() &&
```

Comment:
This can be simplified to:

```return left.getProperties().isPartitionedOnKeys(leftSymbols) && right.getProperties().isPartitionedOnKeys(rightSymbols)```

`isPartitionedOnKeys()` checks `isPartitioned()` internally.